packages/git/src/browser/no-scm-widget.tsx

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ * Copyright (C) 2018 Arm and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, multiInject, postConstruct } from 'inversify';
+import { GitRepositoryProvider } from './git-repository-provider';
+import { ScmWidgetFactory } from './index';
+import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
+import * as React from 'react';
+
+@injectable()
+export abstract class NoScmWidget extends ReactWidget {
+
+    @inject(GitRepositoryProvider)
+    protected readonly repositoryProvider: GitRepositoryProvider;
+
+    constructor(
+        @multiInject(ScmWidgetFactory) public readonly scmWidgetFactories: ScmWidgetFactory[],
```

Comment:
The reason we use contribution providers is to break cyclic dependencies. So it would be better to use them here as well, as it is very easy to run into such cycles.